test/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_service_test.clj

```diff
@@ -398,39 +398,50 @@
             :error))))))
 
 (deftest ^:integration debouncing-test
-  (testing "Default debouncing behavior"
+  (testing "debouncing"
     (let [root-dir (fs/temp-dir "debounce")
           actual-events (atom [])
-          actual-calls (atom 0)
+          callback-invocations (atom 0)
           callback (fn [e]
                      (swap! actual-events concat e)
-                     (swap! actual-calls inc))
+                     (swap! callback-invocations inc))
           app (tk/boot-services-with-config watch-service-and-deps {})
           svc (tk-app/get-service app :FilesystemWatchService)
-          event-cadence (quot watch-core/window-min 2)]
+          event-cadence (/ watch-core/window-min 2)]
       ;; We trigger half the number of events that could independently occur
       ;; without exceeding the window-max, given the fixed event-cadence
       ;; set above, we set the event-cadence to half of window-min to increase
       ;; the likelihood that they will be caught regardless of test runner
-      ;; speed. This requires that window-max be at least four times the
+      ;; speed.
+      ;;
+      ;; TODO what does "be caught" mean?
+      ;;
+      ;; This requires that window-max be at least four times the
       ;; duration of window-min.
-      (testing "multiple events trigger one callback when within `window-min` and `window-max`"
+      (testing "multiple events trigger one callback when within window-min and window-max"
         (let [test-dir (fs/file root-dir "window-min")
-              event-nums (quot (quot watch-core/window-max watch-core/window-min) 2)
-              files (map #(fs/file test-dir (str % ".txt")) (range 0 event-nums))
-              expected-events (set (map (fn [f] {:changed-path f :type :create}) files))]
-          (is (> event-nums 1))
+              ;; TODO shouldn't num-events be (quot (quot window-max event-cadence) 2) ?
+              ;; (based on the the previous comment)
```

Comment:
It would be more likely for us to hit window-max if we used that number. But we still shouldn't hit it under normal circumstances, if you'd prefer it to be calculated like that.

The above number is a direct translation of:
> We trigger half the number of events that could independently occur without exceeding the window-max

We could in theory have one event per window-min that would trigger the code to loop again, so the minimum number of events that could trigger the window-max timeout would be `(quot window-max window-min)`.